feature_tests/steps/result_steps.py

```diff
@@ -0,0 +1,87 @@
+from behave import then
+from domain.core.reference import Reference
+
+from feature_tests.steps.common import parse_value, read_value_from_path
+
+
+@then("the result is of type {type_name}")
+def step_impl(context, type_name):
+    t = type(context.result).__name__
+    assert t == type_name, f"Expected '{type_name}' got '{t}'"
+
+
+@then("the result {field} equals {value:Number}")
+def step_impl(context, field, value):
+    assert context.result is not None, "result"
+    v = context.result.__dict__[field]
+    assert v == value, f"Expected {value} but found {v}"
+
+
+@then('the result {field:String} equals "{value:String}"')
+def step_impl(context, field, value):
+    assert context.result is not None, "result"
+    v = context.result.__dict__[field]
+    assert v == value, f"Expected {value} but found {v}"
+
+
+@then("the result {field:String} equals {value:UUID}")
+def step_impl(context, field, value):
+    assert context.result is not None, "result"
+    v = context.result.__dict__[field]
+    assert v == value, f"Expected {value} but found {v}"
+
+
+@then('the result {field:Path} is a Reference to {system:String} "{value:String}"')
+def step_impl(context, field, system, value):
+    ref = context.result.__dict__[field]
+    assert isinstance(ref, Reference), "Is reference"
+    assert ref.system == system, f"Expected '{system}' got '{ref.system}'"
+    assert ref.value == value, f"Expected '{value}' got '{ref.value}'"
+
+
+def _assert_value(obj, path, value):
+    [head, *tail] = path
+    v = obj.__dict__[head]
+    if len(tail) > 0:
+        _assert_value(v, tail, value)
+    else:
+        assert (
+            v == value
+        ), f"Expected '{type(value).__name__}:{value}' got '{type(v).__name__}:{v}'"
+
+
+@then("the result is")
+def step_impl(context):
```

Comment:
this needs a doc string at the bare minimum - it's not easy to draw a line between what's happening here and what's in the table